docs/subsystems/release-checklist.md

```diff
@@ -112,6 +112,7 @@ preparing a new release.
     - Build a docker image: `cd tools/ci && docker build . -f Dockerfile.prod --build-arg=BASE_IMAGE=zulip/ci:bookworm --build-arg=VERSION=7.0 --tag=zulip/ci:bookworm-7.0 && docker push zulip/ci:bookworm-7.0`
     - Add a new line to the `production_upgrade` matrix in
       `.github/workflows/production-suite.yml`.
+  - Update /history page.
```

Comment:
Specify that this is in `templates/corporate/history.md` to save the work of finding it.